src/sync/sync_session.cpp

```diff
@@ -382,15 +404,17 @@ void SyncSession::nonsync_transact_notify(sync::Session::version_type version)
 
 void SyncSession::revive_if_needed()
 {
-    util::Optional<std::function<SyncLogInHandler>&> log_in_handler;
+    util::Optional<std::function<SyncBindSessionHandler>&> handler;
     {
         std::unique_lock<std::mutex> lock(m_state_mutex);
         if (m_state->revive_if_needed(lock, *this)) {
-            log_in_handler = m_config.log_in_handler;
+            handler = m_config.bind_session_handler;
         }
     }
-    if (log_in_handler)
-        log_in_handler.value()(m_realm_path, m_config);
+    if (handler) {
+        REALM_ASSERT(shared_from_this());
```

Comment:
After doing some research, I realized that this doesn't do what I think it does and I'll remove it. Thanks for pointing it out.

(It was intended as a sanity check, but it's useless since: 1. the way sessions are used is such that they can never exist without being already managed by a `shared_ptr`; 2. if no `shared_ptr` to the session already existed that method wouldn't return null, it would just do something undefined.)